FreeRTOS/FreeRTOS
# Add reconnection and resubscribe logic to the MQTT plaintext demo

**Reconnection logic for the MQTT plaintext demo**

Description
-----------

This implements retries for FreeRTOS IoT Beta2 libraries, initializing the backoff to some value returned by the PRNG: `uxRand()`. If an attempt (connect or subscribe) fails, `vTaskDelay` is called to make the task sleep for some backoff period that doubles on each retry (but is bounded).

Test Steps
-----------

Just run the demo as you normally would.

Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
